packages/lab/src/slider/internal/utils.ts

```diff
@@ -1,158 +1,54 @@
-import { useMemo } from "react";
-import { SliderValue } from "../types";
-import { LabeledMark, SliderMark } from "./SliderRailMarks";
+import { SliderValue, SliderChangeHandler } from "../types";
+import { RefObject } from "react";
 
-const updateValueItemNotPushable = (
-  oldValue: number[],
-  index: number,
-  valueItem: number,
-  min: number,
-  max: number
-) => {
-  const newValue = [...oldValue];
-  if (valueItem < oldValue[index]) {
-    const constraint = index === 0 ? min : newValue[index - 1];
-    newValue[index] = Math.max(constraint, valueItem);
-  } else {
-    const constraint =
-      index === newValue.length - 1 ? max : newValue[index + 1];
-    newValue[index] = Math.min(constraint, valueItem);
-  }
-  return newValue;
-};
-
-const updateValueItemPushable = (
-  oldValue: number[],
-  index: number,
-  valueItem: number,
+export const getValue = (
+  trackRef: RefObject<Element>,
   min: number,
   max: number,
-  pushDistance: number
+  step: number,
+  setValue: SliderChangeHandler,
+  onChange: SliderChangeHandler | undefined,
+  event: MouseEvent
 ) => {
-  const newValue = [...oldValue];
-  newValue[index] = valueItem;
-  if (valueItem < oldValue[index]) {
-    for (let i = index - 1; i >= 0; --i) {
-      if (newValue[i + 1] - newValue[i] < pushDistance) {
-        newValue[i] = newValue[i + 1] - pushDistance;
-      } else {
-        break;
-      }
-    }
-    const distToMin = newValue[0] - min;
-    if (distToMin < 0) {
-      for (let i = index; i >= 0; --i) {
-        newValue[i] -= distToMin;
-      }
-    }
-  } else {
-    for (let i = index + 1; i < newValue.length; ++i) {
-      if (newValue[i] - newValue[i - 1] < pushDistance) {
-        newValue[i] = newValue[i - 1] + pushDistance;
-      } else {
-        break;
-      }
-    }
-    const distToMax = max - newValue[newValue.length - 1];
-    if (distToMax < 0) {
-      for (let i = index; i < newValue.length; ++i) {
-        newValue[i] += distToMax;
-      }
-    }
+  const { clientX } = event;
+  if (trackRef.current) {
+    const { width, x } = trackRef.current.getBoundingClientRect();
+    const localX = clientX - x;
+    const normaliseBetweenValues = (localX / width) * (max - min) + min;
+    let value = roundToStep(normaliseBetweenValues, step);
+    value = roundToTwoDp(value);
```

Comment:
rounding to 2 dp atm - think this makes sense to accommodate currency values - i believe the guidance would be that slider shouldn't really be using decimal places. Can increase the number of dp if we think it is needed ?